enhancements/api-review/azure_user_defined_tags.md

```diff
@@ -0,0 +1,349 @@
+  User-defined tags can be updated on the following Azure resources.
+  - ResourceGroup
+  - Storage Account
+  - DNS Zones
+  - DNS Records
+  - ApplicationSecurityGroup
+  - AvailabilitySet
+  - Group
+  - LoadBalancer
+  - PublicIPAddress
+  - RouteTable
+  - SecurityGroup
+  - VirtualMachineExtension
+  - NetworkInterface
+  - VirtualMachine
+  - VirtualNetwork
+
+- Because tags in `userTags` must be applied to all Azure resources that OpenShift creates,
+  tags must fit all the constraints of the various Azure resources as described below:
+    1. Tag names cannot have `microsoft`, `azure`, `windows` prefixes which are
+       reserved for Azure use.
+    2. An Azure storage account has a limit of 128 characters for the tag name.
+    3. An Azure DNS zone or Traffic or Front Door resource tag name cannot have spaces,
+       special/unicode characters or start with a number.
+    4. An Azure Automation or Content Delivery Network or DNS resource can have a
+       maximum of 15 tags.
+
+- Administrator will have to manually perform below tags pertaining actions
+    1. removing the undesired tags from the required resources.
+    2. update tag values of the required resources.
+    3. update tags of the resources which are not managed by an operator.
+    4. update tags of the resources for which update logic is not supported by an operator.
+
+## Design Details
+
+### Open Questions
+
+### Test Plan
+- Upgrade/downgrade testing
+- Sufficient time for feedback
+- Available by default
+- Stress testing for scaling and tag update scenarios
+
+### Graduation Criteria
+#### Dev Preview -> Tech Preview
+- Feature available for end-to-end usage.
+- Complete end user documentation.
+- UTs and e2e tests are present.
+- Gather feedback from the users.
+
+#### Tech Preview -> GA
+N/A. This feature is for Tech Preview, until decided for GA.
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+On upgrade:
+- Scenario 1: Upgrade to OpenShift version having support for adding tags.
+  The new status field won't be populated since it is only populated by the
+  installer and that can't have happened if the cluster was installed from a
+  prior version. Components that consume the new field should take no action
+  since there won't be any tags to apply.
+- Scenario 2: Upgrade from OpenShift version having support for adding tags to higher:
+  Cluster operators that add tags to Azure resources created for the cluster
+  should refer to the tag fields and take action. For any new resource created
+  post-upgrade, the operator managing the resource will add the user-defined tags
+  to the resource. But the same does not apply to already existing resources,
+  components may or may not update the resources with the user defined tags.
+
+On downgrade:
+- Scenario 1: Cluster installed with OpenShift version having support for adding tags
+  and later downgrading to a lower version not having support for addding tags.
```

Comment:
Typo: "addding".

Is it possible to install one version and then downgrade to an earlier version?

Reply:
comment incorporated.